### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
This is just a bit of cleanup from migrating this code to Java.

The "big" change in here is that _fieldsProto is now {} instead of `null` for empty documents.

### CODE REVIEW

File: src/document.js [hunk 1]
```diff
@@ -576,8 +576,9 @@ class DocumentSnapshot {
    * @return {boolean}
    */
   get isEmpty() {
-    return !this._fieldsProto || isEmptyObject(this._fieldsProto);
+    return this._fieldsProto === undefined || isEmptyObject(this._fieldsProto);
```
Comment (stephenplusplus) [inline src/document.js]: Since we have `is`, how about `is.undefined(this._fieldsProto)`?

Comment (stephenplusplus) [reply src/document.js]: There is also `is.empty(obj)`

File: src/document.js [hunk 2]
```diff
@@ -652,7 +650,6 @@ class DocumentSnapshot {
 
   /**
    * Encodes a JavaScrip value into the Firestore 'Value' representation.
```
Comment (stephenplusplus) [inline src/document.js]: JavaScript*

File: src/document.js (hunk 1 above)
Comment (schmidt-sebastian) [reply src/document.js]: Good suggestions. I removed `isEmptyObject`.

File: src/document.js (hunk 2 above)
Comment (schmidt-sebastian) [reply src/document.js]: Fixed

**Comment by stephenplusplus** [comment]
Linting failed :( Probably another case of your local copy pulling from npm’s cache.

**Comment by schmidt-sebastian** [comment]
> Linting failed :( Probably another case of your local copy pulling from npm’s cache.

Yes, I still don't quite know where that is coming from. I continued on some other PRs as well that have the same errors which I will send out shortly. When I finally figure out the root cause, I'll update them all in one go.

**Comment by stephenplusplus** [comment]
I'll just approve so you're good to merge whenever linting stops being a downer.

**Comment by schmidt-sebastian** [comment]
Whaaaa.... Updating `eslint` and `prettier` to their latest dependencies worked.

**Pull request merged**